### ISSUE TITLE
net-im/telegram-0.9.49

### ISSUE THREAD

**Issue by p-kozlov** [rindeal/gentoo-overlay#36]
https://desktop.telegram.org/#v-0-9-49-16-05-16

Tried to just rename ebuild - didn't help, it's not compiling.

**Comment by rindeal** [comment]
Yeah I know, upstream is pretty wild, its commit messages are worthless as well as the changelog. I'll need to manually search for the changes and implement them. I've already done some, but it's still not finished.

You can track the progress in the [telegram branch](https://github.com/rindeal/gentoo-overlay/tree/telegram) and potentially suggest improvements.

Until I complete the changes, you can use the official pre-compiled binary if you want to try out the new stuff.

**Comment by rindeal** [comment]
I'm the greatest of all, I did it.

https://github.com/rindeal/gentoo-overlay/commit/414b575b4877c8acc64a4873a5cf69b7efc7b162

Please test and report back if any issue occurs.

**Comment by p-kozlov** [comment]
Something is still wrong.
```gawk: /var/db/repos/rindeal/net-im/telegram/files/format_pro.awk:5: fatal: cannot open file `-i' for reading (No such file or directory)```

**Comment by rindeal** [comment]
It's probably because you're using stable gawk which is quite old, gawk-4.1 is now in the process of stabilization, but until then please keyword gawk-4.1, emerge it and try again.

I'm re-opening it now, until I add the missing dependency into the ebuild.

**Comment by p-kozlov** [comment]
Looks like I am unlucky person. Now I am getting another error. After update it requires dev-qt/qt-telegram-static-5.6.0_p20160510 (it was older version before) and now I see this message.
```Downloading 'https://github.com/telegramdesktop/tdesktop/raw/master@{2016-05-10}/Telegram/Patches/qtbase_5_6_0.diff'
--2016-05-27 11:49:00--  https://github.com/telegramdesktop/tdesktop/raw/master@%7B2016-05-10%7D/Telegram/Patches/qtbase_5_6_0.diff
Resolving github.com... 192.30.252.121
Connecting to github.com|192.30.252.121|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2016-05-27 11:49:00 ERROR 404: Not Found.```

**Comment by p-kozlov** [comment]
There is file https://raw.githubusercontent.com/telegramdesktop/tdesktop/master/Telegram/Patches/qtbase_5_6_0.diff there is no `master@{2016-05-10}` branch or tag.

**Comment by rindeal** [comment]
It requires qt-telegram-static-5.6.0_p*, because the newest Telegram now depends on Qt5.6-specific features and doesn't compile with Qt5.5 any more.

I don't know why you're getting that error, because the URL is valid. `master@{2016-05-10}` is a valid date based git revision (`man gitrevisions`).

Are you getting the same error when you put the URL into your browser?

**Comment by p-kozlov** [comment]
Yes, it's fine now. Strange - possibly it was short github outage.

**Comment by rindeal** [comment]
65f9294
